.github/workflows/release.yaml

```diff
@@ -1,25 +1,38 @@
-# This is a template `release.yaml` file for ops charms
-# This file is managed by bootstack-charms-spec and should not be modified
-# within individual charm repos. https://launchpad.net/bootstack-charms-spec
-
 name: Release to Edge
 
 on:
   push:
-    branches: [master, main]
-
-concurrency:
-  group: release
-  cancel-in-progress: true
+    branches: [ main ]
+  release:
+    types: [ published ]
 
 jobs:
   check:
     uses: ./.github/workflows/check.yaml
+    secrets: inherit
 
   release:
     needs: check
-    uses: canonical/bootstack-actions/.github/workflows/charm-release.yaml@v2
-    secrets: inherit
-    with:
-      channel: "latest/edge"
-      upload-image: false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Initialize lxd  # This should dropped once it's implemented on charming-actions itself. https://github.com/canonical/charming-actions/issues/140
+        if: ${{ inputs.destructive-mode == false }}
+        run: |
+          sudo lxd init --auto
+          # The following is needed to fix network error when creating new base instance on lxd
+          sudo chmod a+wr /var/snap/lxd/common/lxd/unix.socket
+          lxc network set lxdbr0 ipv6.address none
+          sudo usermod -a -G lxd $USER
+          if [[ $(cat /etc/os-release | grep VERSION_CODENAME) == *"jammy"* ]]; then
+            sudo iptables -F FORWARD
+            sudo iptables -P FORWARD ACCEPT
+          fi
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.4.0
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          channel: "latest/edge"
+          destructive-mode: false
```

Comment:
could you please add a comment explaining why we want destructive mode to be false?

Reply:
Since we are building for multiple bases with one charmcraft, I think it makes sense to not use destructive mode.
The reason is simply because I want to avoid running snapcraft on older ubuntu to be able to align with what is defined in charmcraft.yaml (run-on). Like this we can avoid need to change `runs-on` on multiple places, to me destructive mode is useful if we're building multiple charm file and releasing them separately.

Does it make sense?

Reply:
ok so if you want to build multiple charms in a single action (and thus single machine), you can use non-destructive mode to build inside lxds. Did I get that right?

Reply:
Yes, if you want to use a single machine per single `build-on` you can use destructive mode. The destructive mode is a little bit faster.

Reply:
I'm ok with this approach although I don't entirely understand the benefit of 1 machine with multiple build-on and non-destructive mode vs one build-on per machine and destructive mode. But please add at least a one-liner comment above line 38 saying why destructive-mode is enabled

Reply:
I'm disabling the destructive mode, which is by default enabled.
The destructive mode can be handy if we decide to build on 20.04 and 22.04 separately, instead of building on one and running on both. To be honest, I'm not sure what I should but in such comment, since I only think that destructive mode is easier for us for now, but maybe we should use it.

Reply:
I check the official documentation for build option for [snapcraft](https://snapcraft.io/docs/build-options) and there is note that we should matched the version of CI env and what is defined in snapcraft.yaml (same for charmcraft [doc](https://juju.is/docs/sdk/charmcraft-pack)), so that's why I want to avoid it, because we can easily forget to change one.

Reply:
thanks for the note, now I understand that `destructive-mode` is actually not the default when you build via the cli (odd that the action defaults to true)